lib/tests/test_lexer.c

```diff
@@ -156,7 +156,7 @@ _current_lloc(void)
 #define assert_parser_float(expected)                           \
   _next_token();                                                        \
   assert_token_type(LL_FLOAT);                                         \
-  assert_true(_current_token()->fnum == expected, "Bad parsed value at %s:%d", __FUNCTION__, __LINE__);
+  assert_true(_current_token()->fnum >= expected, "Bad parsed value at %s:%d", __FUNCTION__, __LINE__);
```

Comment:
It is not enough to check if the `expected` is smaller than the `fnum`, it should check if the distance between the two number are so small that it is insignificant.

Reply:
ok. We can compare the difference like,
```
assert_true((_current_token()->fnum - expected) < 1e-12, "Bad parsed value at %s:%d", __FUNCTION__, __LINE__);

```

Reply:
Just like the example in the 3rd comment, you should use `fabs`.

Reply:
ok. Thanks @Kokan